src/app/loadout-builder/mod-utils.ts

```diff
@@ -42,11 +42,21 @@ export function someModHasEnergyRequirement(mods: PluggableInventoryItemDefiniti
   );
 }
 
+/**
+ * This is used to create a string representation of each mod for the purpose of
+ * creating permutations when we are calculating assignments.
+ *
+ * The reason we do this instead of using say the hash, is that if we have 5 general
+ * mods that all cost 1 energy and have energy type Any, we would generate 1
+ * permutation of that set, as opposed to 5! if we used hashes and they were all
+ * unique.
+ */
```

Comment:
Afraid I still don't follow this - I don't get what the string looks like or what it has to do with permutations or why you can generate one instead of five.

Reply:
Ok I will try and explain it better. It will require an understanding of how heaps algorithm work though. I could have two versions of that and lose the generics if you think it makes it easier to understand.